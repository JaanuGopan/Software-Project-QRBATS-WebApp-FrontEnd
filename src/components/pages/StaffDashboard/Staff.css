.staff-container {
    margin: 20px;
    height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.cards {
    display: flex;
    justify-content: center;
    /* Center the cards horizontally */
    flex-wrap: wrap;
    gap: 20px;
    width: 100%;
    /* Ensure the cards take full width */
}

.card {
    border-radius: 5px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;
    border-radius: 30px;
}

.card:hover {
    transform: translateY(-10px);
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.3);
}

.top {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 10px;
    color: white;
    padding: 20px;
    font-size: 30px;
}

.icon {
    padding-top: 20px;
    padding-left: 50px;
    width: 20px;
    height: 20px;
    font-size: 60px;
    padding-right: 50px;
}

.label {
    font-weight: bold;
    color: white;
    padding: 30px;
    font-size: 20px;
    padding-right: 90px;
}

.bottom {
    padding: 10px;
}

.heading {
    font-size: 24px;
    font-weight: bold;
    margin-bottom: 10px;
    padding-bottom: 30px;
    padding-top: 30px;
}

.buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    padding-top: 20px;

}

.create-button,
.delete-button {
    display: flex;
    align-items: center;
    padding: 10px;
    background-color: #0e7ac8;
    color: white;
    cursor: pointer;
    border-radius: 15px;
}

.create-button:hover,
.delete-button:hover {
    background-color: #095e8e;
}

.create-button svg,
.delete-button svg {
    margin-right: 10px;
}