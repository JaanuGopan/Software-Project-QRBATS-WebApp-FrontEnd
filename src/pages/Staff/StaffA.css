@media screen and (max-width: 1800px) {
  .staff-mainHead {
    font-size: x-large;
    margin-top: 5px;
    padding-left: 35px;
  }
  .staff-buttonIcon {
    height: 1.5rem;
    width: 1.5rem;
    size: 1.5rem;
  }
  .staff-Dash {
    margin-top: 20px;
  }
  .staff-SearchEvent {
    margin-right: 35px;
  }
}

@media screen and (max-width: 768px) {
  .staff-mainHead {
    font-size: large;
    margin-top: 10px;
    padding-left: 35px;
  }
  .staff-buttonIcon {
    height: 1rem;
    width: 1rem;
    size: 1rem;
  }
  .staff-Dash {
    margin-top: 10px;
  }
  .staff-SearchEvent {
    margin-right: 35px;
  }
}

@media screen and (max-width: 426px) {
  .staff-buttonIcon {
    height: 0.5rem;
    width: 0.5rem;
    size: 0.5rem;
  }
  .staff-Dash {
    margin-top: 5px;
  }
  .staff-SearchEvent {
    margin-right: 0px;
  }
  .staff-mainHead {
    padding-left: 5px;
  }
}

.staff-Dash {
  flex-grow: 1;
  height: 85vh;
  width: 100%;
}

.staff-mainHead {
  color: #0d295e;
  font-weight: bold;
  margin-bottom: 0px;
}

.staff-EventList {
  display: flex;
  flex-direction: column;
  margin-top: 10px;
  padding-left: 10px;
  padding-right: 10px;
  padding: 10px;
  width: 100%;
  height: 78vh;
  border-top: 5px solid #0365a1;
  box-shadow: 0.5px 2px 0.5px 0.5px rgba(25, 65, 85, 0.5);
  border-radius: 10px;
  background-color: #fafafa;
}

.staff-List-Buttons {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-around;
  padding: 1rem;
}

.staff-Create-Event-Dashboard {
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  overflow-y: scroll;
  width: 100%;
  height: 100vh;
  padding-top: 5vh;
  padding-bottom: 5vh;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.5);
}

.staff-SearchEvent {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  padding-right: 10px;
}

.staff-signup-main-container {
  position: relative;
  width: 50%;
  border: 4px solid transparent;
  border-top: 6px solid #0077c0;
  border-radius: 10px;
  padding: 20px;
  background-color: white;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  min-width: 250px;
}

.staff-head1 {
  text-align: center;
  color: black;
  font-weight: bold;
  font-size: x-large;
  padding: 1%;
  margin: 0%;
}

.staff-login-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  margin-top: 20px;
  position: relative;
}

.staff-image-container {
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.staff-logo {
  width: 200px;
  height: auto;
  border-radius: 10px;
}

.staff-select-input {
  border-radius: 10px;
  word-wrap: normal;
  padding: 5px;
  width: 100%;
  border-bottom: 3px solid #0063a0;
}

.staff-creation-input {
  display: flex;
  flex-direction: row;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  align-self: center;
  justify-self: center;
  margin: auto;
  margin-top: 1%;
  margin-bottom: 1%;
  padding-left: 5%;
  padding-right: 5%;
}

.staff-creation-input label {
  width: 30%;
}

.staff-creation-input-field {
  width: 60%;
}

.create-staff-create-button {
  width: 50%;
  align-self: center;
  justify-self: center;
  margin: auto;
  margin-top: 20px;
}

.create-staff-create-button button {
  width: 100%;
}

.staff-update-title-close-button {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.staff-update-title {
  align-self: center;
  justify-self: center;
  margin-left: auto;
}

.staff-update-close-button {
  align-self: center;
  justify-self: flex-end;
  margin-left: auto;
  height: 30px;
  width: 30px;
  align-items: center;
  justify-content: center;
}

.staff-update-close-button #close-icon {
  align-self: center;
  justify-self: center;
  color: red;
  height: 90%;
  width: 90%;
}

.staff-update-close-button #close-icon:hover {
  align-self: center;
  justify-self: center;
  color: white;
  height: 90%;
  width: 90%;
  background-color: red;
  border-radius: 100%;
}
