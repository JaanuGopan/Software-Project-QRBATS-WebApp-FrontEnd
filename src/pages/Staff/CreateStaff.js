import React, { useState } from "react";
import "./StaffA.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateUserService from "../../api/services/CreateUserService";

const CreateStaff = ({ handlecloseCreateStaffWindow }) => {
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [userName, setUserName] = useState("");
  const [userRole, setUserRole] = useState("");
  const [departmentId, setDepartmentId] = useState("");

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = CreateUserService.saveUser(
        firstname,
        lastname,
        email,
        password,
        userName,
        deparmentList.indexOf(departmentId) + 1,
        userRoleList.indexOf(userRole)
      );

      console.error("Login success...");
      handlecloseCreateStaffWindow();
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  return (
    <div className="staff-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateStaffWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="staff-head1">Create Staff</p>
      <div className="staff-login-container">
        <div className="staff-image-container">
          <img src={Designer} className="staff-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="choice-input mb-3">
              <select
                value={userRole}
                onChange={(e) => setUserRole(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select user role</option>
                {userRoleList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />
            <InputField
              placeholder="Confirm password"
              value={confirmPassword}
              onChange={(e) => setConfirmPassword(e.target.value)}
              inputType="password"
            />

            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Add Staff
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStaff;
