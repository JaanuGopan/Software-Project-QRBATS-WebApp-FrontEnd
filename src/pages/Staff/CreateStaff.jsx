import React, { useState } from 'react';
import './StaffA.css';
import InputField from '../../components/textfields/InputBox/InputField';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import CreateUserService from '../../api/services/CreateUserService';
import Select from 'react-select';
import { toast } from 'react-toastify';
import { CircularProgress } from '@mui/material';
import Department from '../../utils/Department';

const CreateStaff = ({ handleCloseCreateStaffWindow, reloadStaffList }) => {
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [userName, setUserName] = useState('');
  const [userRole, setUserRole] = useState('');
  const [departmentId, setDepartmentId] = useState('');

  const departmentList = Department.departmentList;
  const userRoleList = ['ADMIN', 'LECTURER'];

  const notifySuccess = () => toast.success('Successfully Staff Created!');

  const [loadingCreateStaff, setLoadingCreateStaff] = useState(false);

  const handleInputValidation = () => {
    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    if (firstName.trim() === '') {
      toast.error('Please enter first name');
      return false;
    }
    if (lastName.trim() === '') {
      toast.error('Please enter last name');
      return false;
    }
    if (email.trim() === '' || !emailPattern.test(email)) {
      toast.error('Please enter a valid email');
      return false;
    }
    if (userName.trim() === '') {
      toast.error('Please enter username');
      return false;
    }
    if (password.trim() === '') {
      toast.error('Please enter password');
      return false;
    }
    if (confirmPassword.trim() === '') {
      toast.error('Please confirm password');
      return false;
    }
    if (password !== confirmPassword) {
      toast.error('Passwords do not match');
      return false;
    }
    if (!departmentId) {
      toast.error('Please select department');
      return false;
    }
    if (!userRole) {
      toast.error('Please select user role');
      return false;
    }
    return true;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!handleInputValidation()) {
      return;
    }
    try {
      setLoadingCreateStaff(true);
      const response = await CreateUserService.saveUser(
        firstName,
        lastName,
        email,
        password,
        userName,
        departmentList.indexOf(departmentId.value) + 1,
        userRoleList.indexOf(userRole.value)
      );
      if (response.status === 200) {
        notifySuccess();
        reloadStaffList();
        handleCloseCreateStaffWindow();
        toast.success('User created successfully!');
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch (error) {
      console.error('User creation failed', error);
      toast.error('Error in user creation.');
    } finally {
      setLoadingCreateStaff(false);
    }
  };

  return (
    <div className="staff-signup-main-container">
      <div className="staff-update-title-close-button">
        <h3 className="staff-update-title">Create User</h3>
        <div className="staff-update-close-button" onClick={handleCloseCreateStaffWindow}>
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>

      <div className="staff-login-container">
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="staff-creation-input">
              <label>User Role</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={userRole}
                  onChange={(e) => setUserRole(e)}
                  options={userRoleList.map((role) => ({
                    value: role,
                    label: role,
                  }))}
                  placeholder={'Select User Role'}
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>First Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your first name"
                  value={firstName}
                  onChange={(e) => setFirstName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Last Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your last name"
                  value={lastName}
                  onChange={(e) => setLastName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="staff-creation-input">
              <label>Email</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>User Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your user name"
                  value={userName}
                  onChange={(e) => setUserName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="staff-creation-input">
              <label>Password</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your password"
                  value={password}
                  onChange={(e) => setPassword(e.target.value)}
                  inputType="password"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Confirm Password</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Confirm password"
                  value={confirmPassword}
                  onChange={(e) => setConfirmPassword(e.target.value)}
                  inputType="password"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Department</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={departmentId}
                  onChange={(e) => setDepartmentId(e)}
                  options={departmentList.map((dept) => ({
                    value: dept,
                    label: dept,
                  }))}
                  placeholder={'Select Department'}
                />
              </div>
            </div>
            <div className="create-staff-create-button">
              {loadingCreateStaff ? (
                <div className="d-flex justify-content-center align-items-center">
                  <CircularProgress />
                </div>
              ) : (
                <button type="submit" className="btn btn-success">
                  Create Staff
                </button>
              )}
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStaff;
