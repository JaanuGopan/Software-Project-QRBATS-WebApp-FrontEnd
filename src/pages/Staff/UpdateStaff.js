import React, { useState } from "react";
import "./StaffA.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import { useNavigate } from "react-router-dom";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";

const UpdateStaff = ({
  handlecloseUpdateStaffWindow,
  selectedStaff,
  handleReloadStaffList,
}) => {
  const [userId, setUserId] = useState(selectedStaff.userId);
  const [firstname, setFirstName] = useState(selectedStaff.firstName);
  const [lastname, setLastName] = useState(selectedStaff.lastName);
  const [email, setEmail] = useState(selectedStaff.email);
  const [password, setPassword] = useState("");
  const [userName, setUserName] = useState(selectedStaff.username);
  const [userRole, setUserRole] = useState(selectedStaff.role);
  const [departmentId, setDepartmentId] = useState(selectedStaff.departmentId);
  const navigate = useNavigate();

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER", "STAFF"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await axios.post(
        "http://localhost:8080/api/v1/auth/signin",
        {
          userId: userId,
          firstName: firstname,
          lastName: lastname,
          email: email,
          password: password,
          userName: userName,
          departmentId: deparmentList.indexOf(departmentId) + 1,
        }
      );
      const token = response.data.token;
      const decodedToken = parseJwt(token);
      const userName = decodedToken.sub;
      localStorage.setItem("token", token);
      navigate("/mainNavigation", { state: { userName } });
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  const parseJwt = (token) => {
    const base64Url = token.split(".")[1];
    const base64 = base64Url.replace(/-/g, "+").replace(/_/g, "/");
    const jsonPayload = decodeURIComponent(
      atob(base64)
        .split("")
        .map((c) => "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2))
        .join("")
    );
    return JSON.parse(jsonPayload);
  };

  return (
    <div className="staff-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseUpdateStaffWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="staff-head1">Update Staff Details</p>
      <div className="staff-login-container">
        <div className="staff-image-container">
          <img src={Designer} className="staff-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />
            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="staff-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Save
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default UpdateStaff;
