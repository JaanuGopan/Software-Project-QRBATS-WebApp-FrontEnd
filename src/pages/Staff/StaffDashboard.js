import React, { useState, useEffect } from "react";
import "./StaffA.css";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import CreateStaff from "./CreateStaff";
import UpdateStaff from "./UpdateStaff";
import StaffTable from "../../components/layout/AdminDashboardComponent/StaffTable";
import FetchStaffService from "../../api/services/FetchStaffService";
import DeleteStaffService from "../../api/services/DeleteStaffService";

const StaffDashboard = () => {
  const [staffCreatePopUpWindow, setStaffCreatePopUpWindow] = useState(false);
  const [staffUpdatePopUpWindow, setStaffUpdatePopUpWindow] = useState(false);
  const [search, setSearch] = useState("");

  const [staffsList, setStaffsList] = useState([]);
  const [selectedStaff, setSelectedStaff] = useState(null);

  useEffect(() => {
    // Fetch the list of staffs from the API
    handleReloadStaffList();
  }, []);

  const handleEventClick = (staffs) => {
    setSelectedStaff(staffs);
    // Do whatever you want with the selected event data
    console.log("Selected Staffs:", staffs);
  };

  const handleDelete = async () => {
    try {
      const response = await DeleteStaffService.deleteStaff(
        selectedStaff.userId
      );
      handleReloadStaffList();
    } catch (error) {
      console.log("error " + error);
    }
  };

  const handleReloadStaffList = async () => {
    FetchStaffService.fetchStaffs()
      .then((staffs) => {
        setStaffsList(staffs);
      })
      .catch((error) => {
        console.error("Error fetching staffs:", error);
      });
  };

  return (
    <div className="staff-Dash">
      <div className="staff-SearchEvent">
        <p className="staff-mainHead">Staff Details</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: "150px",
            padding: "3px 40px",
            border: "0.5px solid black",
            borderRadius: "5px",
            textAlign: "center",
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="staff-EventList">
        <StaffTable
          search={search}
          handleUpdateStaff={() => setStaffUpdatePopUpWindow(true)}
          onStaffClick={handleEventClick}
          staffsList={staffsList}
        />
        <div className="staff-List-Buttons">
          <NormalButton
            handleClick={() => setStaffCreatePopUpWindow(true)}
            title={"Create"}
            titlewithiconicon={
              <MdCreateNewFolder className="staff-buttonIcon" />
            }
          />
          <NormalButton
            title={"Delete"}
            handleClick={handleDelete}
            titlewithiconicon={
              <RiDeleteBin5Fill className="staff-buttonIcon" />
            }
          />
        </div>
      </div>
      {staffCreatePopUpWindow && (
        <div
          handleClick={() => setStaffCreatePopUpWindow(false)}
          className="staff-Create-Event-Dashboard"
        >
          <CreateStaff
            handlecloseCreateStaffWindow={() =>
              setStaffCreatePopUpWindow(false)
            }
          />
        </div>
      )}
      {staffUpdatePopUpWindow && (
        <div
          handleClick={() => setStaffCreatePopUpWindow(false)}
          className="staff-Create-Event-Dashboard"
        >
          <UpdateStaff
            handlecloseUpdateStaffWindow={() =>
              setStaffUpdatePopUpWindow(false)
            }
            selectedStaff={selectedStaff}
            handleReloadStaffList={handleReloadStaffList}
          />
        </div>
      )}
    </div>
  );
};

export default StaffDashboard;
