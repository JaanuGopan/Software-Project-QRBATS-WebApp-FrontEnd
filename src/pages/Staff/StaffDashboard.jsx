import React, { useState, useEffect } from 'react';
import './StaffA.css';
import NormalButton from '../../components/layout/AdminDashboardComponent/NormalButton';
import { MdCreateNewFolder } from 'react-icons/md';
import { RiDeleteBin5Fill } from 'react-icons/ri';
import CreateStaff from './CreateStaff';
import UpdateStaff from './UpdateStaff';
import StaffTable from '../../components/layout/AdminDashboardComponent/StaffTable';
import DeleteStaffService from '../../api/services/DeleteStaffService';
import { toast, ToastContainer } from 'react-toastify';
import WarningPopup from '../../components/warningPopup/WarningPopup';
import AdminService from '../../api/services/AdminService';

const StaffDashboard = () => {
  const [staffCreatePopUpWindow, setStaffCreatePopUpWindow] = useState(false);
  const [staffUpdatePopUpWindow, setStaffUpdatePopUpWindow] = useState(false);
  const [search, setSearch] = useState('');
  const [staffsList, setStaffsList] = useState([]);
  const [selectedStaff, setSelectedStaff] = useState(null);

  useEffect(() => {
    // Fetch the list of staffs from the API
    handleReloadStaffList();
  }, []);

  const handleEventClick = (staffs) => {
    setSelectedStaff(staffs);
    // Do whatever you want with the selected event data
    console.log('Selected Staffs:', staffs);
  };

  const [showDeletePopUpWindow, setShowDeletePopUpWindow] = useState(false);

  const [processingDeleteStaff, setProcessingDeleteStaff] = useState(false);

  const handleDelete = async () => {
    if (!selectedStaff) {
      toast.error('Please select a staff to delete');
      return;
    }
    try {
      setProcessingDeleteStaff(true);
      const response = await DeleteStaffService.deleteStaff(selectedStaff.userId);
      if (response.status === 200) {
        toast.success('Staff Deleted Successfully');
        setSelectedStaff(null);
        handleReloadStaffList();
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch (error) {
      console.log('error ' + error);
      toast.error('Error In Staff Deletion. ');
    } finally {
      setProcessingDeleteStaff(false);
      setShowDeletePopUpWindow(false);
    }
  };

  const handleReloadStaffList = async () => {
    AdminService.fetchStaffs()
      .then((staffs) => {
        setStaffsList(staffs);
      })
      .catch((error) => {
        console.error('Error fetching staffs:', error);
      });
  };

  return (
    <div className="staff-Dash">
      <ToastContainer />
      <div className="staff-SearchEvent">
        <p className="staff-mainHead">Staff Details</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: '150px',
            padding: '3px 40px',
            border: '0.5px solid black',
            borderRadius: '5px',
            textAlign: 'center',
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="staff-EventList">
        <StaffTable
          search={search}
          handleUpdateStaff={() => setStaffUpdatePopUpWindow(true)}
          onStaffClick={handleEventClick}
          staffsList={staffsList}
        />
        <div className="staff-List-Buttons">
          <NormalButton
            handleClick={() => setStaffCreatePopUpWindow(true)}
            title={'Create'}
            titlewithiconicon={<MdCreateNewFolder className="staff-buttonIcon" />}
          />
          <NormalButton
            title={'Delete'}
            handleClick={() => {
              if (selectedStaff == null) {
                toast.error('Please Select Staff. ');
              } else {
                setShowDeletePopUpWindow(true);
              }
            }}
            titlewithiconicon={<RiDeleteBin5Fill className="staff-buttonIcon" />}
          />
        </div>
      </div>
      {staffCreatePopUpWindow && (
        <div className="staff-Create-Event-Dashboard">
          <CreateStaff
            handleCloseCreateStaffWindow={() => setStaffCreatePopUpWindow(false)}
            reloadStaffList={handleReloadStaffList}
          />
        </div>
      )}
      {staffUpdatePopUpWindow && (
        <div
          handleClick={() => setStaffCreatePopUpWindow(false)}
          className="staff-Create-Event-Dashboard"
        >
          <UpdateStaff
            handleCloseUpdateStaffWindow={() => setStaffUpdatePopUpWindow(false)}
            selectedStaff={selectedStaff}
            handleReloadStaffList={handleReloadStaffList}
          />
        </div>
      )}
      {showDeletePopUpWindow && selectedStaff && (
        <div className="student-creation-delete-popup-window">
          <WarningPopup
            handleOk={handleDelete}
            handleCloseWarningWindow={() => setShowDeletePopUpWindow(false)}
            buttonText={'Delete'}
            titleText={'Are you sure you want to delete this staff?'}
            processing={processingDeleteStaff}
          />
        </div>
      )}
    </div>
  );
};

export default StaffDashboard;
