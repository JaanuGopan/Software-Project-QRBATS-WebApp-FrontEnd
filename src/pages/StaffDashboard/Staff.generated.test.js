import renderer from 'react-test-renderer';
import React, { Component } from "react";
import { RiUserLine } from "react-icons/ri";

import { FaUsers, FaEdit, FaTrashAlt, FaPlus } from "react-icons/fa";
import Staff from "./Staff";

jest.mock("react-icons/ri");
jest.mock("./Staff.css");
jest.mock("react-icons/fa");

const renderTree = tree => renderer.create(tree);
describe('<Staff>', () => {
  it('should render component', () => {
    expect(renderTree(<Staff 
    />).toJSON()).toMatchSnapshot();
  });
  
});