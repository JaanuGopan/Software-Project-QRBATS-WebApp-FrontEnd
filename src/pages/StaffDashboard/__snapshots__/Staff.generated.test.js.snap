// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Staff> should render component 1`] = `
<div
  className="staff-container"
>
  <h2>
    Staff
  </h2>
  <div
    className="cards"
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#0e7ac8",
        }
      }
    >
      <div
        className="top"
      >
        <div
          className="icon"
        />
        <div
          className="bottom"
        >
          5
        </div>
      </div>
      <div
        className="label"
      >
        Lecturer
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#0e7ac8",
        }
      }
    >
      <div
        className="top"
      >
        <div
          className="icon"
        />
        <div
          className="bottom"
        >
          7
        </div>
      </div>
      <div
        className="label"
      >
        Normal Staff
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#0e7ac8",
        }
      }
    >
      <div
        className="top"
      >
        <div
          className="icon"
        />
        <div
          className="bottom"
        >
          3
        </div>
      </div>
      <div
        className="label"
      >
        Admin
      </div>
    </div>
  </div>
  <div
    className="table"
  >
    <div
      className="heading"
    >
      Normal Staff Details
    </div>
    <table>
      <thead>
        <tr>
          <th>
            No
          </th>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            Username
          </th>
          <th>
            Password
          </th>
          <th>
            Edit
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            John Doe
          </td>
          <td>
            john.doe@example.com
          </td>
          <td>
            johndoe
          </td>
          <td>
            ****
          </td>
          <td />
        </tr>
        <tr>
          <td>
            1
          </td>
          <td>
            John Doe
          </td>
          <td>
            john.doe@example.com
          </td>
          <td>
            johndoe
          </td>
          <td>
            ****
          </td>
          <td />
        </tr>
      </tbody>
    </table>
    <div
      className="buttons"
    >
      <button
        className="create-button"
      >
         Create Lecturer Account
      </button>
      <button
        className="delete-button"
      >
         Delete
      </button>
    </div>
  </div>
</div>
`;
